frontend/src/editor/automerge_websocket_editor.ts

```diff
@@ -17,17 +17,16 @@ enum MessageSyncType {
 
 export function useAutomergeWebsocketEditor(
   url: string | URL,
-  { onInitialSyncComplete }: { onInitialSyncComplete: () => void },
-): Editor {
+  { onInitialSyncComplete }: { onInitialSyncComplete: (editor?: Editor) => void },
+): [Editor?, Paragraph[]?] {
   const debug = useDebugMode();
-
-  const editor = useMemo(() => {
-    const baseEditor = createEditor();
-    const editorWithReact = withReact(baseEditor);
-    const editorWithAutomerge = withAutomergeDoc(editorWithReact, Automerge.init());
-    return editorWithAutomerge;
-  }, [url.toString()]);
-
+  const [editorAndInitialValue, setEditorAndInitialValue] = useState<null | {
```

Comment:
The idea behind making these two one state is, that changing only one of them does not work. So to prevent one from accidentially changing only one and expecting it to work I combined them to more clearly signal this behaviour.